client/components/account-balances/utils.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { timeOfDayStrings } from './strings';
+
+/**
+ * Calculates the time of day based on the browser's time.
+ *
+ * @param {Date} date A date object to calculate the time of day for. Defaults to the current time.
+ * @return {string} The time of day (translatable string). One of 'morning', 'afternoon' or 'evening'.
+ *
+ */
+export const getTimeOfDayString = ( date: Date = new Date() ): string => {
+	const hour = date.getHours();
+	// Morning 5am -11.59am
+	if ( hour >= 5 && hour < 12 ) {
+		return timeOfDayStrings.morning;
+	}
+	// Afternoon 12pm – 4:59pm
+	if ( hour >= 12 && hour < 17 ) {
+		return timeOfDayStrings.afternoon;
+	}
+	// Evening 5pm – 4:59am
+	return timeOfDayStrings.evening;
+};
+
+/**
+ * Returns a greeting string based on the time of day and a given name, if provided.
+ *
+ * @param {string} name An name to include in the greeting, optional.
+ * @param {Date} date A date object to calculate the time of day for. Defaults to the current time.
+ * @return {string} A greeting string.
+ */
+export const getGreeting = (
+	name?: string,
+	date: Date = new Date()
+): string => {
+	const timeOfDay = getTimeOfDayString( date );
+	const adjective = __( `Good `, 'woocommerce-payments' );
+	let greeting = `${ adjective }${ timeOfDay }`;
+	if ( name ) {
+		greeting += `, ${ name }`;
+	}
+	greeting += ' 👋';
```

Comment:
Just a pre-submittal heads up. When translating strings they should be translated as a complete sentence. 

ie 
```php
// Placeholder is user's name. 
__( 'Good evening, %s', 'woocommerce-payments' );
```


This is because some languages need to change the sentence structure. eg in Polish it's "dzień dobry Jamesie" (Day good, James) or "Good" on it's own may be translated into 1 word but a different word in this context (greeting).

Reply:
This somewhat complicates things because you've got the user's name being conditionally appended. 

But you could change the `timeOfDayStrings` into full greetings. eg `Good morning, %', 'Good afternoon, %s', 'Good evening, %s'.

and if conditional user names are something we need to account for, I can only think we then need another set of greetings without username placeholds. 🙈

Reply:
Thanks for the heads up @james-allan 🙌

Reply:
I've changed these strings to remain as entire phrases in ef916e4